src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -72,3 +75,7 @@ void WaveformMarkSet::clear() {
     m_defaultMark = WaveformMark();
     m_marks.clear();
 }
+
+WaveformMark& WaveformMarkSet::getHotCueMark(int hotCue) {
```

Comment:
Returning a pointer to internal data structures is not recommended, either.

Better use a setter that overwrites/replaces the entire marker if 
WaveformMark is copyable: setHotCueMark(int hotCue, const WaveformMark& marker)

Don't sacrifice efficiency over safety ;)

On 18.08.2016 08:22, Ferran Pujol Camins wrote:
>
> In src/waveform/renderers/waveformmarkset.cpp 
> <https://github.com/mixxxdj/mixxx/pull/992#discussion_r75253843>:
>
> > @@ -72,3 +75,7 @@ void WaveformMarkSet::clear() {
> >      m_defaultMark = WaveformMark();
> >      m_marks.clear();
> >  }
> > +
> > +WaveformMark& WaveformMarkSet::getHotCueMark(int hotCue) {
>
> What's different from returning a pointer?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/992/files/fbcc455724d82073cdee184087bb7cdf53684167#r75253843>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-arTlyPL0yFpUwW1cTYlnycZeUz7Rks5qg_oegaJpZM4Jj7vm>.
>